integration/class-wp-rest-cache.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * WP REST Cache integration file.
+ *
+ * This file contains code for caching ActivityPub REST requests.
+ *
+ * Copyright (C) 2025 Epiphyt
+ * Original code: https://epiph.yt/en/blog/2025/accidental-ddos-through-activitypub-plugin/
+ *
+ * Portions of this code are adapted from GPL v2 licensed code.
+ * As such, you may also redistribute and/or modify those portions under the terms of
+ * the GNU General Public License as published by the Free Software Foundation.
+ *
+ * https://www.gnu.org/licenses/gpl-2.0.html
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Integration;
+
+use Activitypub\Collection\Outbox;
+use Activitypub\Comment;
+use WP_Rest_Cache_Plugin\Includes\Caching\Caching;
+
+/**
+ * Compatibility with the WP REST Cache plugin.
+ *
+ * @see https://wordpress.org/plugins/wp-rest-cache/
+ */
+class WP_Rest_Cache {
+	/**
+	 * Initialize the class, registering WordPress hooks.
+	 */
+	public static function init() {
+		\add_filter( 'wp_rest_cache/allowed_endpoints', array( self::class, 'add_activitypub_endpoints' ) );
+		\add_filter( 'wp_rest_cache/determine_object_type', array( self::class, 'set_object_type' ), 10, 4 );
+		\add_filter( 'wp_rest_cache/is_single_item', array( self::class, 'set_is_single_item' ), 10, 3 );
+		\add_action( 'transition_post_status', array( self::class, 'transition_post_status' ), 10, 3 );
+		\add_action( 'transition_comment_status', array( self::class, 'transition_comment_status' ), 10, 3 );
+	}
+
+	/**
+	 * Add ActivityPub endpoints to the list of allowed endpoints.
+	 *
+	 * @param array $endpoints List of allowed endpoints.
+	 *
+	 * @return array
+	 */
+	public static function add_activitypub_endpoints( $endpoints ) {
+		$endpoints[ ACTIVITYPUB_REST_NAMESPACE ] = array( 'actors', 'collections', 'comments', 'interactions', 'nodeinfo', 'posts', 'users' );
```

Comment:
isn't the namespace simply `activitypub` or `activitypub/1.0`?

Reply:
Yes, `activitypub/1.0`

Reply:
I meant, can we wildcard all the subpathes?